### DISCUSSION TITLE
Self Hosted - Closed Registration - Invite Function?

### DISCUSSION THREAD

**Discussion by danielharrelson** [simple-login/app#551] [Feature Requests]
I have a 3 character TLD, and always used it for email. I'm self hosting, and don't want to have it open to the public for registrations, however, it would be nice to be able to send invites to friends and family members who would like to utilize the service/app on my domain for privacy.

Is there any way to do this, aside from just opening registration, and then immediently closing it once they have an account?

**Comment by nguyenkims** [comment]
@danielharrelson one way to do it is to add authentication to Nginx, for example https://docs.nginx.com/nginx/admin-guide/security-controls/configuring-http-basic-authentication/

This way you can restrict access to only people who have the password.

**Reply by danielharrelson** [reply]
Yes, I thought about that. However, since I have my API running on https://[domain].[tld] that would jack up the browser extensions, no?

**Reply by danielharrelson** [reply]
I guess I could use my API as https://[user]:[pass]@[domain].[tld]/ ?

**Reply by nguyenkims** [reply]
I think that could work but I never tried this before.

**Reply by danielharrelson** [reply]
Just tested it, it works for Firefox API calls, however it does not work on IOS app.

**Reply by nguyenkims** [reply]
Oh thanks for the info. A workaround can be ignore the Nginx authentication on the API urls, which always starts by `/api/`. This is just a workaround as one person can use the API to sign up for new accounts.

**Reply by danielharrelson** [reply]
This is acceptable, as I figure most people won't know what I'm running behind the URL to post API calls.

Example config for those wondering:
```
location / {
        proxy_pass http://localhost:7777;
        auth_basic "Login Required";
        auth_basic_user_file /etc/nginx/simple.htpasswd;
}

location ^~ /api/ {
        proxy_pass http://localhost:7777/api/;
}
```
